src/master/heartbeat_cluster.cpp

```diff
@@ -471,6 +464,176 @@ namespace cubhb
     return valid_ping_host_exists;
   }
 
+  void
+  cluster::handle_heartbeat (const heartbeat_arg &arg, ipv4_type from_ip)
+  {
+    if (shutdown || m_hostname != arg.get_dest_hostname ())
+      {
+	// if the cluster is stopped or there is a mismatch on hostname then exit
+	return;
+      }
+
+    pthread_mutex_lock (&lock);
+
+    // apply heartbeat request on unidentified host entries
+    apply_heartbeat_on_ui_node (arg, from_ip);
```

Comment:
hb_cluster_receive_heartbeat used to check:
```c
      hb_state = (cubhb::node_entry::node_state) state;	

       if (hb_state < cubhb::node_entry::UNKNOWN || hb_state >= cubhb::node_entry::MAX)	
	{	
	  MASTER_ER_LOG_DEBUG (ARG_FILE_LINE, "receive heartbeat have unknown state. (state:%u).\n", state);	
	  pthread_mutex_unlock (&hb_Cluster->lock);	
	  return;	
	}
```

Is state still checked?

Reply:
I added a `assert (m_state >= node_state::UNKNOWN && m_state < node_state::MAX)` when unpacking the state